snapcraft/tests/test_yaml.py

```diff
@@ -346,6 +347,107 @@ def test_config_expands_filesets(self, mock_loadPlugin):
         })
 
 
+class TestYamlEnvironment(tests.TestCase):
+
+    def setUp(self):
+        super().setUp()
+        dirs.setup_dirs()
+
+        tempdirObj = tempfile.TemporaryDirectory()
+        self.addCleanup(tempdirObj.cleanup)
+        os.chdir(tempdirObj.name)
```

Comment:
We already have this on the base setup, when this test runs the cwd is already a tempdir that will be deleted. Tha path to that dir is in self.path.
https://github.com/ubuntu-core/snapcraft/blob/master/snapcraft/tests/__init__.py#L33